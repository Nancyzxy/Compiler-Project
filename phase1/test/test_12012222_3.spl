int main(){
    float _a = 2.2;
    int a_ = 2;
    while(a_ < 6){
        if(_a > 8.8){
            _a = (_a + a_ * 8)/5 + 22;  
        }
        a_ = a_ + 1;
    }
    return 0;
}