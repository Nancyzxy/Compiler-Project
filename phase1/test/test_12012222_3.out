Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: main
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: float
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: _a
                ASSIGN
                Exp (2)
                  FLOAT: 2.2
            SEMI
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: a_
                  ASSIGN
                  Exp (3)
                    INT: 2
              SEMI
        StmtList (4)
          Stmt (4)
            WHILE
            LP
            Exp (4)
              Exp (4)
                ID: a_
              LT
              Exp (4)
                INT: 6
            RP
            Stmt (4)
              CompSt (4)
                LC
                StmtList (5)
                  Stmt (5)
                    IF
                    LP
                    Exp (5)
                      Exp (5)
                        ID: _a
                      GT
                      Exp (5)
                        FLOAT: 8.8
                    RP
                    Stmt (5)
                      CompSt (5)
                        LC
                        StmtList (6)
                          Stmt (6)
                            Exp (6)
                              Exp (6)
                                ID: _a
                              ASSIGN
                              Exp (6)
                                Exp (6)
                                  Exp (6)
                                    LP
                                    Exp (6)
                                      Exp (6)
                                        ID: _a
                                      PLUS
                                      Exp (6)
                                        Exp (6)
                                          ID: a_
                                        MUL
                                        Exp (6)
                                          INT: 8
                                    RP
                                  DIV
                                  Exp (6)
                                    INT: 5
                                PLUS
                                Exp (6)
                                  INT: 22
                            SEMI
                        RC
                  StmtList (8)
                    Stmt (8)
                      Exp (8)
                        Exp (8)
                          ID: a_
                        ASSIGN
                        Exp (8)
                          Exp (8)
                            ID: a_
                          PLUS
                          Exp (8)
                            INT: 1
                      SEMI
                RC
          StmtList (10)
            Stmt (10)
              RETURN
              Exp (10)
                INT: 0
              SEMI
        RC
